### PR TITLE
replace validate_string with datatype

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/actionpolicy/rule.pp [hunk 1]
```diff
@@ -1,13 +1,12 @@
 # Define - mcollective::actionpolicy::rule
 define mcollective::actionpolicy::rule(
-  $agent,
-  $action      = 'allow',
-  $callerid    = '*',
-  $actions     = '*',
-  $fact_filter = '*',
-  $classes     = '*'
+  String $agent,
+  String $action      = 'allow',
```
Comment (vinzent) [inline manifests/actionpolicy/rule.pp]: isn't action more an Enum ?

Comment (vinzent) [inline manifests/actionpolicy/rule.pp]: is even something other than allow allowed here? https://github.com/puppetlabs/mcollective-actionpolicy-auth#usage

Comment (bastelfreak) [inline manifests/actionpolicy/rule.pp]: I'm not sure which values are allowed here. Do you have a list?

Comment (vinzent) [inline manifests/actionpolicy/rule.pp]: https://github.com/puppetlabs/mcollective-actionpolicy-auth#policy-file-format

```
Any number of policy lines are permitted. These must be tab delimited lines with either four or five fields (the final field is optional) in the following order:

    allow or deny
```

Comment (bastelfreak) [inline manifests/actionpolicy/rule.pp]: @vinzent fixed

File: examples/ssl_example/mco_profile/manifests/middleware/activemq.pp [hunk 2]
```diff
@@ -68,9 +68,9 @@
     trustcacerts => true,
     notify       => Class['activemq::service'],
     require      => File["${confdir}/ca.pem"],
-  } ->
+  }
 
-  file { "${confdir}/truststore.jks":
+  -> file { "${confdir}/truststore.jks":
```
Comment (vinzent) [inline examples/ssl_example/mco_profile/manifests/middleware/activemq.pp]: why moving them?

why specifying them at all? puppet4 has manifest ordering by default.

File: manifests/client.pp [hunk 3]
```diff
@@ -9,6 +9,6 @@
   contain ::mcollective::client::install
   contain ::mcollective::client::config
 
-  Class['mcollective::client::install'] ->
-  Class['mcollective::client::config']
+  Class['mcollective::client::install']
+  -> Class['mcollective::client::config']
```
Comment (vinzent) [inline manifests/client.pp]: Why moving?

From my perspective the old style is nicer.

File: examples/ssl_example/mco_profile/manifests/middleware/activemq.pp (hunk 2 above)
Comment (vinzent) [inline examples/ssl_example/mco_profile/manifests/middleware/activemq.pp]: ah this is the new puppet-lint s**t. never mind.

Comment (vinzent) [inline examples/ssl_example/mco_profile/manifests/middleware/activemq.pp]: still leaves the why specifying the relation ship at all. :)

File: manifests/client.pp (hunk 3 above)
Comment (bastelfreak) [inline manifests/client.pp]: we agreed somewhere in the IRC to go with the new puppet-lint style. This is recommended by the style guide. We should discuss the style guide if we don't like it, but not disable the linter.

**Pull request merged**